File: public/dsa/arrays/even_number_of_digits/digits_counting/even_number_of_digits.cpp
// imports-start
#include <bits/stdc++.h>
using namespace std;
// imports-end

int getDigitCount(int n) {
    int count = 0;
    if (n == 0) {
        return 1;
    }
    while (n > 0) {
        count++;
        n /= 10;
    }
    return count;
}
vector<int> getEvenDigitNumbers(vector<int> &arr) {
    // add your logic here
    vector<int> result;
    for (int v : arr) {
        if (getDigitCount(v) % 2 == 0) {
            result.push_back(v);
        }
    }
    return result;
}

// tests-start
int main() {
    vector<int> arr = {42, 564, 5775, 34, 123, 454, 1, 5, 45, 3556, 23442};
    auto result = getEvenDigitNumbers(arr);
    for (int v : result) {
        cout << v << " ";
    }
    return 0;
}
// tests-end

/*output
42 5775 34 45 3556
*/

File: public/dsa/arrays/even_number_of_digits/digits_counting/even_number_of_digits.py
from typing import List, Optional


def get_digits_count(v: int):
    if v == 0:
        return 1
    count = 0
    while v > 0:
        v //= 10
        count += 1
    return count


class Solution:
    def getEvenDigitNumbers(self, arr: List[int]) -> Optional[List[int]]:
        result = []
        for v in arr:
            digits_count = get_digits_count(v)
            if digits_count % 2 == 0:
                result.append(v)
        return result


# tests-start
arr = [42, 564, 5775, 34, 123, 454, 1, 5, 45, 3556, 23442]
sol = Solution()
print(sol.getEvenDigitNumbers(arr))
# tests-end

'''output
[42, 5775, 34, 45, 3556]
'''
